actpool/actpool.go

```diff
@@ -317,17 +386,14 @@ func (ap *actPool) GetGasCapacity() uint64 {
 }
 
 func (ap *actPool) Validate(ctx context.Context, selp action.SealedEnvelope) error {
-	ap.mutex.RLock()
-	defer ap.mutex.RUnlock()
 	return ap.validate(ctx, selp)
 }
 
 func (ap *actPool) DeleteAction(caller address.Address) {
-	ap.mutex.RLock()
-	defer ap.mutex.RUnlock()
-	pendingActs := ap.accountActs[caller.String()].AllActs()
-	ap.removeInvalidActs(pendingActs)
-	delete(ap.accountActs, caller.String())
+	worker := ap.worker[ap.allocatedWorker(caller)]
```

Comment:
`worker` is guaranteed `!= nil`?

Reply:
`ap.worker` is a slice, and func `allocatedWorker` assure the worker is always available.